server/catgenome/src/main/java/com/epam/catgenome/manager/FileManager.java

```diff
@@ -1915,17 +1915,19 @@ public void deleteFeatureFileDirectory(FeatureFile featureFile) throws IOExcepti
      * @throws IOException
      */
     public List<AbstractFsItem> loadDirectoryContents(String path) throws IOException {
-        if (!filesBrowsingAllowed) {
+
+        if(!StringUtils.isEmpty(path) && !path.startsWith(ngsDataRootPath)) {
```

Comment:
Is `ngsDataRootPath` always terminated by path separator (`/`)? 
If not, then the following case will allow access:
```
ngsDataRootPath = '/data/storage'
path = '/data/storagesecret'
```

Reply:
@mzueva that is right. 
Fixed.
Now we check it with startsWith method from Path class, that will perform checking as paths, not just strings.